monix-reactive/shared/src/test/scala/monix/reactive/internal/operators/ThrottleLatestSuite.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2014-2021 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.reactive.internal.operators
+
+import monix.execution.Ack.Continue
+import monix.reactive.{Observable, Observer}
+import monix.reactive.internal.operators.MaxSuite.{assert, assertEquals, test}
+
+import scala.concurrent.duration._
+
+object ThrottleLatestSuite extends BaseOperatorSuite {
+  def sum(sourceCount: Int) = {
+    sourceCount * (sourceCount - 1) / 2
+  }
+  def createObservable(sourceCount: Int) = Some {
+    if (sourceCount == 1) {
+      val o = Observable.now(100L).delayExecution(500.millis).throttleLatest(1.second, true)
+      Sample(o, 1, 100, 500.millis, 1.second)
+    } else {
+
+      val o = Observable
+        .intervalAtFixedRate(2.second)
+        .take(sourceCount.toLong)
+        .throttleLatest(1.second, true)
```

Comment:
If the rate of output is at 2 seconds and you're throttling by 1 second after first element (as per RxJava doc you linked), isn't that a noop?

Reply:
Yeah, probably true. I will think about something, but as far as I know we can't control the exact elements that are going to be throttled in this test, same as in `SampleOnce`.

Reply:
You should be able to do something like
```scala
Observable(1).delayOnNext(200.millis) ++ Observable(2).delayOnNext(100.millis) ++ ...
```
to create a predictable chain.

Reply:
I'm also concerned about `Sample` two lines below, as that is literally saying it's not dropping anything from the input (2nd parameter of sample is expected output size). The purpose of throttling is to drop some so we should check this if possible. I think ThrottleFirstSuite has an example where it drops half.

Reply:
Yeah, that is correct. I was mostly basing on `SampleOnceSuite` here. No worries though I will fix it now :)

Reply:
> 

You should be able to do something like

Observable(1).delayOnNext(200.millis) ++ Observable(2).delayOnNext(100.millis) ++ ...

to create a predictable chain.

Not sure what's the difference between that and just doing the `intervalAtFixedRate` to be honest :)

Reply:
I think something like that can be useful if we would like to have a specific test case in a test outside `BaseOperatorSuite`, e.g. [ReactiveX example](http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#throttleLatest-long-java.util.concurrent.TimeUnit-)